src/codegen/mod.rs

```diff
@@ -2172,6 +2177,206 @@ impl CodeGenerator for CompInfo {
+        let ret_type = return_item.to_rust_ty_or_opaque(ctx, &());
+        let args = signature.argument_types();
+        let second_arg_type = if args.len() < 2 {
+            None
+        } else {
+            dbg!("test");
+            let temp = utils::type_id_to_rust_type(ctx, signature.argument_types()[1].1);
+            Some(raw_pointer_to_reference(temp))
+        };
+
+        // We then check if the function name is in one of the following three
+        // arrays or "operator==". If yes, we generate the correct wrapper and
+        // add it to result.
+        let assignment_operators = [
+            ("operator+=", "AddAssign", "add_assign"),
+            ("operator&=", "BitAndAssign", "bitand_assign"),
+            ("operator|=", "BitOrAssign", "bitor_assign"),
+            ("operator^=", "BitXorAssign", "bitxor_assign"),
+            ("operator/=", "DivAssign", "div_assign"),
+            ("operator%=", "RemAssign", "rem_assign"),
+            ("operator<<=", "ShlAssign", "shl_assign"),
+            ("operator>>=", "ShrAssign", "shr_assign"),
+            ("operator-=", "SubAssign", "sub_assign"),
+        ];
+        let binary_operators = [
+            ("operator+", "Add", "add"),
+            ("operator&", "BitAnd", "bitand"),
+            ("operator|", "BitOr", "bitor"),
+            ("operator^", "BitXor", "bitxor"),
+            ("operator/", "Div", "div"),
+            ("operator%", "Rem", "rem"),
+            ("operator<<", "Shl", "shl"),
+            ("operator>>", "Shr", "shr"),
+            ("operator-", "Sub", "sub"),
+        ];
+        let unary_operators = [
+            ("operator-", "Neg", "neg"),
+            ("operator!", "Not", "not"),
+        ];
+        if name == "operator==" {
+            result.push(quote!(
+                impl PartialEq for #ty_for_impl {
+                    fn eq(&self, rhs: &Self) -> bool {
+                        unsafe {
+                            #function_name(self, rhs)
+                        }
+                    }
+                }
+            ));
+            return;
+        }
+        for el in assignment_operators.iter() {
+            // The args.len() == 2 check shouldn't be neccessary but we still have it.
+            if args.len() == 2 && name == el.0 {
+                let rhs_type = second_arg_type.unwrap();
+                let trait_name = proc_macro2::TokenStream::from_str(el.1).unwrap();
+                let func_name = proc_macro2::TokenStream::from_str(el.2).unwrap();
+                result.push(quote!(
+                    impl std::ops::#trait_name<#rhs_type> for #ty_for_impl {
+                        fn #func_name(&mut self, rhs: #rhs_type) {
+                            let retptr = unsafe {
+                                #function_name(self, rhs)
+                            };
+                            assert_eq!(retptr, self as *mut #ty_for_impl, "The bindgen authors have no idea how to handle this case.");
```

Comment:
I don't think we necessarily care about asserting this here, do we? As in, the rust trait doesn't have any return value, it's fine to drop it on the floor.

Reply:
Where C++ thinks the object lives and where Rust thinks the object lives have to be identical. Most C++ operators just return this; , but afaik this is not required by the standard.

Reply:
Right, my point is that where the object lives is not meaningful is it? Like, a C++ operator can return whatever it wants, but it can't destroy `this`. So I still don't know why this assert is useful.

Reply:
I removed this assertion in the newest commit. I hope you are right...